src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/boss_anetheron.cpp

```diff
@@ -37,232 +37,124 @@ enum Texts
+    void JustSummoned(Creature* summon) override
     {
-        npc_towering_infernalAI(Creature* creature) : ScriptedAI(creature)
+        if (summon)
         {
-            instance = creature->GetInstanceScript();
-            AnetheronGUID = instance->GetGuidData(DATA_ANETHERON);
+            summon->AI()->DoCast(SPELL_INFERNAL_STUN);
+            summon->SetInCombatWithZone();
         }
+        BossAI::JustSummoned(summon);
+    }
 
-        uint32 ImmolationTimer;
-        uint32 CheckTimer;
-        ObjectGuid AnetheronGUID;
-        InstanceScript* instance;
+    void DoAction(int32 action) override
+    {
+        Talk(SAY_ONSPAWN, 1200ms);
 
-        void Reset() override
-        {
-            DoCast(me, SPELL_INFERNO_EFFECT);
-            ImmolationTimer = 5000;
-            CheckTimer = 5000;
-        }
+        if (action == DATA_ANETHERON)
+            me->GetMotionMaster()->MovePath(urand(ALLIANCE_BASE_CHARGE_1, ALLIANCE_BASE_CHARGE_3), false);
+    }
 
-        void JustEngagedWith(Unit* /*who*/) override
+    void PathEndReached(uint32 pathId) override
+    {
+        switch (pathId)
         {
+        case ALLIANCE_BASE_CHARGE_1:
+        case ALLIANCE_BASE_CHARGE_2:
+        case ALLIANCE_BASE_CHARGE_3:
+            me->m_Events.AddEventAtOffset([this]()
+                {
+                    me->GetMotionMaster()->MovePath(urand(ALLIANCE_BASE_PATROL_1, ALLIANCE_BASE_PATROL_3), true);
+                }, 1s);
+            break;
         }
+    }
 
-        void KilledUnit(Unit* /*victim*/) override
+    void KilledUnit(Unit* victim) override
+    {
+        if (!_recentlySpoken && victim->IsPlayer())
         {
-        }
+            Talk(SAY_ONSLAY);
+            _recentlySpoken = true;
 
-        void JustDied(Unit* /*killer*/) override
-        {
+            scheduler.Schedule(6s, [this](TaskContext)
+                {
+                    _recentlySpoken = false;
+                });
         }
+    }
 
-        void MoveInLineOfSight(Unit* who) override
-
-        {
-            if (me->IsWithinDist(who, 50) && !me->IsInCombat() && me->IsValidAttackTarget(who))
-            {
-                AttackStart(who);
-            }
-        }
+    void JustDied(Unit * killer) override
+    {
+        Talk(SAY_ONDEATH);
+        BossAI::JustDied(killer);
+    }
 
-        void UpdateAI(uint32 diff) override
-        {
-            if (CheckTimer <= diff)
-            {
-                if (AnetheronGUID)
-                {
-                    Creature* boss = ObjectAccessor::GetCreature(*me, AnetheronGUID);
-                    if (!boss || boss->isDead())
-                    {
-                        me->setDeathState(DeathState::JustDied);
-                        me->RemoveCorpse();
-                        return;
-                    }
-                }
-                CheckTimer = 5000;
-            }
-            else CheckTimer -= diff;
+    void UpdateAI(uint32 diff) override
```

Comment:
this method is already in BossAI and thus does not need to be overwritten if nothing specific is added. So you can remove this